common/src/main/java/pl/edu/agh/utils/MessageUtils.java

```diff
@@ -1,12 +1,59 @@
 package pl.edu.agh.utils;
 
-import io.netty.buffer.ByteBuf;
-import org.apache.commons.lang3.SerializationUtils;
+import pl.edu.agh.messages.RaftMessage;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Base64;
 
 public class MessageUtils {
-    public static Object toObject(ByteBuf byteBuf) {
-        byte[] bytes = new byte[byteBuf.readableBytes()];
-        byteBuf.getBytes(0, bytes);
-        return SerializationUtils.deserialize(bytes);
+    private static final int MAX_PARTS = 4;
```

Comment:
I had to make some kind of backoff. Without it MessageUtils would sometimes merge hundreds of subsequent messages not being able to deserialize them